src/main/java/com/sleekbyte/tailor/Tailor.java

```diff
@@ -247,53 +264,74 @@ public static CommonTokenStream getTokenStream(File inputFile) throws IOExceptio
-                listeners.add(new MaxLengthListener(printer, constructLengths, enabledRules));
-                listeners.add(new MinLengthListener(printer, constructLengths, enabledRules));
-                DeclarationListener decListener = new DeclarationListener(listeners);
-                listeners.add(decListener);
-
-                ParseTreeWalker walker = new ParseTreeWalker();
-                for (SwiftBaseListener listener : listeners) {
-                    // The following listeners are used by DeclarationListener to walk the tree
-                    if (listener instanceof ConstantNamingListener || listener instanceof KPrefixListener) {
-                        continue;
+        List<File> files = fileNames.parallelStream().map(File::new).collect(Collectors.toList());
+        List<Optional<CommonTokenStream>> tokenStreams =
+            files.parallelStream().map(Tailor::getTokenStream).collect(Collectors.toList());
+        System.out.format("Analyzing %s:%n", pluralize(fileNames.size(), "file", "files"));
+        List<Optional<TopLevelContext>> trees = tokenStreams
+            .parallelStream()
+            .map(stream -> {
+                    Optional<TopLevelContext> tree = Optional.empty();
+                    try {
+                        tree = Tailor.getParseTree(stream);
+                        System.out.print(".");
+                    } catch (ErrorListener.ParseException e) {
+                        System.out.print("S");
+                        numSkippedFiles++;
                     }
-                    walker.walk(listener, tree);
-                }
-                try (FileListener fileListener = new FileListener(printer, inputFile, constructLengths, enabledRules)) {
-                    fileListener.verify();
+                    return tree;
+                })
+            .collect(Collectors.toList());
+        System.out.format("%n");
+        ListIterator<Optional<CommonTokenStream>> tokenStreamIterator = tokenStreams.listIterator();
+        ListIterator<Optional<TopLevelContext>> treeIterator = trees.listIterator();
+        files.parallelStream().forEachOrdered(inputFile -> {
+                Optional<CommonTokenStream> optTokenStream = tokenStreamIterator.next();
+                Optional<TopLevelContext> optTree = treeIterator.next();
+                try {
+                    try (Printer printer = new Printer(inputFile, maxSeverity, colorSettings)) {
+                        if (optTokenStream.isPresent() && optTree.isPresent()) {
+                            CommonTokenStream tokenStream = optTokenStream.get();
+                            TopLevelContext tree = optTree.get();
+                            List<SwiftBaseListener> listeners = createListeners(enabledRules, printer, tokenStream);
+                            listeners.add(new MaxLengthListener(printer, constructLengths, enabledRules));
```

Comment:
do you think we can move all the listener code into another function? there's already a lot happening in this function.